### PR TITLE
Refactoring Reading

### PR DESCRIPTION
**Pull request by DanaK270**
this is what we have done until now:
- defining interfaces for the different types of questions 
- hard-coding a reading test to ensure that it complies with our interfaces.

### CODE REVIEW

**Comment by mzhubail** [comment]
All in all seems to be good.

Next step is to copy over the view code (html) of each question type one by one.

File: packages/frontend/src/components/Reading/ListSelectionQuestionComponent.tsx [hunk 1]
```diff
@@ -1,9 +1,57 @@
...
 }) => {
-  return <p>{JSON.stringify(question)}</p>;
+  // Initialize state to hold the selection for each sub-question as an array of arrays to be in the same format as the tableCompletion component
+  const [selections, setSelections] = useState<string[][]>(
+    question.SubQuestions.map(() => ['']),
+  );
+
+  const generateSelectName = (rowIndex: number) => {
+    return `select-${questionIndex}-${rowIndex}`;
+  };
```
Comment (mzhubail) [inline packages/frontend/src/components/Reading/ListSelectionQuestionComponent.tsx:18]: I don't think we need this.

File: packages/frontend/src/components/Reading/QuestionsComponent.tsx [hunk 2]
```diff
@@ -16,19 +16,29 @@ export const QuestionsComponent = ({
...
+        />
+      );
     case 'List Selection':
-      return <ListSelectionQuestionComponent question={question} />;
+      return (
+        <ListSelectionQuestionComponent
+          question={question}
+          questionIndex={index}
+        />
+      );
```
Comment (mzhubail) [inline packages/frontend/src/components/Reading/QuestionsComponent.tsx]: Same comment as `generateSelectName`

**Review by mzhubail — COMMENTED** [review]
Good work, keep it up!

**Comment by mzhubail** [comment]
@DanaK270 If you go to the reading page you'll see all the parts.  Note that they all now use the same state, which then can be flattened and sent to the back-end.

For now, the questions are still hard-coded.  This shouldn't be an issue for now as another part will be responsible for fetching the data from the backend (e.g. the full mock test websocket).

I think this is resolved for now.

Next step is to create a a proper ui for this.

**Review by Ma-hawaj — APPROVED** [review]
Approved on behalf of @mzhubail

**Pull request merged**